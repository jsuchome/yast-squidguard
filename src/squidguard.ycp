/* ------------------------------------------------------------------------------
 * Copyright (c) 2006 Novell, Inc. All Rights Reserved.
 *
 *
 * This program is free software; you can redistribute it and/or modify it under
 * the terms of version 2 of the GNU General Public License as published by the
 * Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program; if not, contact Novell, Inc.
 *
 * To contact Novell about this file by physical or electronic mail, you may find
 * current contact information at www.novell.com.
 * ------------------------------------------------------------------------------
 */

/**
 * File:	clients/squidguard.ycp
 * Package:	Configuration of squidguard
 * Summary:	Main file
 * Authors:	Lars Vogdt <lars@linux-schulserver.de>
 *
 * $Id: squidguard.ycp 27914 2006-02-13 14:32:08Z locilka $
 *
 * Main file for squidguard configuration. Uses all other files.
 */

{

/***
 * <h3>Configuration of squidguard</h3>
 */

textdomain "squidguard";

/* The main () */
y2milestone ("----------------------------------------");
y2milestone ("Squidguard module started");

import "Progress";
import "Report";
import "Summary";

import "CommandLine";
include "squidguard/wizards.ycp";

map cmdline_description = $[
    "id" 	: "squidguard",
    /* Command line help text for the Xsquidguard module */
    "help"	: _("Configuration of squidguard"),
    "guihandler"        : SquidguardSequence,
    "initialize"        : Squidguard::Read,
    "finish"            : Squidguard::Write,
    "actions"           : $[
	// FIXME TODO: fill the functionality description here
    ],
    "options"		: $[
	// FIXME TODO: fill the option descriptions here
    ],
    "mappings"		: $[
	// FIXME TODO: fill the mappings of actions and options here
    ]
];

/* is this proposal or not? */
boolean propose = false;
list args = WFM::Args();
if(size(args) > 0) {
    if(is(WFM::Args(0), path) && WFM::Args(0) == .propose) {
        y2milestone("Using PROPOSE mode");
        propose = true;
    }
}

/* main ui function */
any ret = nil;

if(propose) ret = SquidguardAutoSequence();
else ret = CommandLine::Run(cmdline_description);
y2debug("ret=%1", ret);

/* Finish */
y2milestone("Squidguard module finished");
y2milestone("----------------------------------------");

return ret;

/* EOF */
}
