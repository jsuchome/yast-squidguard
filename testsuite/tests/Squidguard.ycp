{

// testedfiles: Squidguard.ycp

include "testsuite.ycp";
TESTSUITE_INIT([], nil);

import "Squidguard";

DUMP("Squidguard::Modified");
TEST(``(Squidguard::Modified()), [], nil);

}
