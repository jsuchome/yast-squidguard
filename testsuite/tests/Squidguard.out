Dump	Squidguard::Modified
Return	false
